### PR TITLE
Marian Auth

### PR DESCRIPTION
**Pull request by marianphillips**

### CODE REVIEW

File: src/client/App.jsx [hunk 1]
```diff
@@ -11,6 +11,7 @@ function App() {
   const blank = { username: "", password: "" };
 
   const [register, setRegister] = useState(blank);
+  const [registerMessage, setRegisterMessage] = useState("");
```
Comment (KinTale) [inline src/client/App.jsx:14]: nice use of states

File: src/server/middleware/movie.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const jwt = require('jsonwebtoken')
+const secret = "jwtSecret"
+
+const checkToken = (req, res, next) => {
+    const auth = req.headers.authorization
+
+    if(!auth) {
+        res.status(401)
```
Comment (usamalodhi) [inline src/server/middleware/movie.js:8]: good use of appropriate status codes throughout the exercise

File: src/server/controllers/movie.js [hunk 3]
```diff
@@ -15,6 +15,12 @@ const createMovie = async (req, res) => {
 
     const { title, description, runtimeMins } = req.body;
 
+    if(!title || !description || !runtimeMins) {
```
Comment (ben-corbin) [inline src/server/controllers/movie.js:18]: Like this input verification.

File: src/server/controllers/user.js [hunk 4]
```diff
@@ -9,6 +9,12 @@ const errorMessage = 'Invalid username or password.'
  const registerUser = async (req, res) => {
     const { username, password } = req.body;
 
+    if(!username || !password) {
+        res.status(401)
+        res.json({error: "All fields required"})
+        return
+    }
+
     const hashedPassword = bcrypt.hashSync(password, 10)
```
Comment (ben-corbin) [inline src/server/controllers/user.js:18]: Nice use of hashSync to get your required hash synchronously rather than returning the promise.

Comment (marianphillips) [reply src/server/controllers/user.js:18]: Tbh I only used it becuase thats what seemed to work last time. I like the way you did it more (with await) but didnt want to mess it up by refactoring ha

File: src/server/controllers/movie.js (hunk 3 above)
Comment (KinTale) [reply src/server/controllers/movie.js:18]: agreed. its something i should check for too!

**Pull request closed without merging**